### ISSUE TITLE
[v1alpha1] Machines with replaced instances do not re-join the cluster

### ISSUE THREAD

**Issue by liztio** [kubernetes-sigs/cluster-api-provider-aws#1052]
/kind bug

**What steps did you take and what happened:**
1. Create a v1a1 CAPA management cluster
2. Add a workload cluster
3. Apply an HA cluster (I used the one from `cmd/clusterctl/examples/aws/out/controlplane-machines-ha`)
4. Wait for the cluster to be ready
5. Delete one of the instances for the cluster in the AWS console
6. Wait for the instance to be replaced (The reconcile loop runs every 10 minutes)
7. List the nodes in the workload cluster:

```
NAME                         STATUS   ROLES    AGE   VERSION
ip-10-0-0-235.ec2.internal   Ready    master   79m   v1.14.4
ip-10-0-0-34.ec2.internal    Ready    master   81m   v1.14.4
```

**What did you expect to happen:**
After the instance booted, the machine should've joined the workload cluster 

**Anything else you would like to add:**
cloud-init logs on the new machine: https://gist.github.com/liztio/7c83cf99af52b9eee96c3dba20985709

**Environment:**

- Cluster-api-provider-aws version: Custom image on built on commit `b250c6100d3c0872bfaafda115eef01ba9dd4a96`
- Kubernetes version: (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:54Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-17T00:58:35Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`): `18.04.2 LTS (Bionic Beaver)`

**Comment by ncdc** [comment]
We actually need to mark the Machine as permanently failed, rather than recreating a new EC2 instance.

**Comment by ncdc** [comment]
Right @detiber? xref https://github.com/kubernetes-sigs/cluster-api/issues/1205

**Comment by detiber** [comment]
@ncdc for v1alpha2, yes. I suspect there is some wiggle room for how we handle it for v1alpha1.

**Comment by ncdc** [comment]
For alpha1, do you want to regen a bootstrap token & let the replacement instance join?

**Comment by detiber** [comment]
Considering the additional implications if we do this on a Machine that is part of the control plane, I would likely say it is safer to make it a hard failure similar to v1alpha2, otherwise we would need to differentiate control plane vs worker node.

**Comment by joonas** [comment]
Given this is targeted at v1alpha1, I believe the milestone should be updated to [v0.3.x (v1alpha1)](https://github.com/kubernetes-sigs/cluster-api-provider-aws/milestone/10) instead of [v0.4.0 (v1alpha2)](https://github.com/kubernetes-sigs/cluster-api-provider-aws/milestone/9)

**Comment by ncdc** [comment]
Now that v0.4 is out, we are not really maintaining v0.3 any more. This has been addressed in v0.4 as a permanent failure condition.

/close
